apollo-router/src/spec/query.rs

```diff
@@ -94,6 +94,11 @@ impl Query {
 
         let parser = apollo_parser::Parser::new(string.as_str());
         let tree = parser.parse();
+
+        // Trace log recursion limit data
+        let recursion_limit = tree.recursion_limit();
+        tracing::trace!(?recursion_limit, "recursion limit data");
```

Comment:
Should `recursion_limit` be inlined here to prevent the execution of `recursion_limit()` when tracing is not enabled?
Same goes for the other change in schema.rs.